tests/tests/test-author-mapping.ts

```diff
@@ -54,15 +54,14 @@ describeDevMoonbeam("Author Mapping - simple association", (context) => {
       genesisAccount,
       context.polkadotApi.tx.authorMapping.addAssociation(bobAuthorId)
     );
-
     // check events
-    expect(events.length === 6);
-    expect(context.polkadotApi.events.balances.Reserved.is(events[0])).to.be.true;
-    expect(context.polkadotApi.events.authorMapping.AuthorRegistered.is(events[1])).to.be.true;
-    expect(context.polkadotApi.events.system.NewAccount.is(events[2])).to.be.true;
-    expect(context.polkadotApi.events.balances.Endowed.is(events[3])).to.be.true;
-    expect(context.polkadotApi.events.treasury.Deposit.is(events[4])).to.be.true;
-    expect(context.polkadotApi.events.system.ExtrinsicSuccess.is(events[5])).to.be.true;
+    expect(events.length === 8);
+    expect(context.polkadotApi.events.balances.Reserved.is(events[1])).to.be.true;
+    expect(context.polkadotApi.events.authorMapping.AuthorRegistered.is(events[2])).to.be.true;
+    expect(context.polkadotApi.events.system.NewAccount.is(events[4])).to.be.true;
+    expect(context.polkadotApi.events.balances.Endowed.is(events[5])).to.be.true;
+    expect(context.polkadotApi.events.treasury.Deposit.is(events[6])).to.be.true;
+    expect(context.polkadotApi.events.system.ExtrinsicSuccess.is(events[7])).to.be.true;
```

Comment:
What are events 0 and 3? Should they be tested as well?

Reply:
Perhaps. It seems that the event reordering was due to additional `Withdraw` and `Deposit` events that weren't emitted previously.

I'm suggesting that we refactor at least some of the event assertion code to use some [not-yet-existent] utility functions like `expectEventsToInclude(SomeEvent)`, etc. Most of the testing doesn't seem to be concerned with the exhaustive contents and ordering of events, just that they include a specific event or two.

Alternatively, another improvement would be to assert the entire array at once (this would probably also require a new util fn) such that the error produced when it fails shows the expected vs. actual arrays.